--- CMakeLists.txt
+++ CMakeLists.txt
@@ -221,6 +221,8 @@
 #            NAMES fst
 #            PATHS ${CMAKE_CURRENT_SOURCE_DIR}/tools/openfst/lib
 #            REQUIRED)
+set(OPENFST_ROOT_DIR @OPENFST_ROOT@)
+include(third_party/openfst_lib_target)
 #find_path(OpenFST_INCLUDE_DIR
 #            NAMES "fst/fst.h"
 #            PATHS "${CMAKE_CURRENT_SOURCE_DIR}/tools/openfst/include"
--- cmake/third_party/openfst_lib_target.cmake
+++ cmake/third_party/openfst_lib_target.cmake
@@ -29,3 +29,19 @@
 unset(fst_source_dir)
 unset(fst_include_dir)
 unset(fst_sources)
+
+# Build ngram extension library (needed by Vosk).
+set(fst_ngram_source_dir ${OPENFST_ROOT_DIR}/src/extensions/ngram)
+file(GLOB fst_ngram_sources "${fst_ngram_source_dir}/*.cc")
+add_library(fstngram ${fst_ngram_sources})
+add_dependencies(fstngram fst)
+target_include_directories(fstngram PUBLIC
+     $<BUILD_INTERFACE:${OPENFST_ROOT_DIR}/src/include>
+     $<INSTALL_INTERFACE:include/openfst>
+)
+install(TARGETS fstngram
+    EXPORT kaldi-targets
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
